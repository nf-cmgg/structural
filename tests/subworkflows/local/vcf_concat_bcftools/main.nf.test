nextflow_workflow {

    name "Test Workflow VCF_CONCAT_BCFTOOLS"
    script "subworkflows/local/vcf_concat_bcftools/main.nf"
    workflow "VCF_CONCAT_BCFTOOLS"

    tag "subworkflows"
    tag "subworkflows_local"
    tag "vcf_concat_bcftools"

    config "./nextflow.config"

    test("homo_sapiens") {

        when {
            workflow {
                """
                input[0] = Channel.of([
                    [id:"test", sample:"test"],
                    file(params.sv_vcf1, checkIfExists:true),
                    file(params.sv_tbi1, checkIfExists:true)
                ], [
                    [id:"test", sample:"test"],
                    file(params.cnv_vcf1, checkIfExists:true),
                    file(params.cnv_tbi1, checkIfExists:true)
                ], [
                    [id:"test", sample:"test"],
                    file(params.repeats_vcf1, checkIfExists:true),
                    file(params.repeats_tbi1, checkIfExists:true)
                ])
                input[1] = 3
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    path(workflow.out.vcfs[0][1]).linesGzip[-4..-1],
                    file(workflow.out.vcfs[0][2]).name,
                    workflow.trace.succeeded().size(),
                    workflow.trace.failed().size()
                ).match() }
            )
        }

    }

}
