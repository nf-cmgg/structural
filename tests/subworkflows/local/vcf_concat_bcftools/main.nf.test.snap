{
    "homo_sapiens": {
        "content": [
            [
                "chrX\t15000\t.\tA\t.\t.\tLowDepth\tEND=45826;REPREF=15413;RL=30826;RU=CA;VARID=CNBP_CA;REPID=CNBP_CA\tGT:SO:REPCN:REPCI:ADSP:ADFL:ADIR:LC\t.:.:.:.:.:.:.:0",
                "chrX\t1030000\tqdnaseq_67\tN\t<DUP>\t.\tPASS\tCNV_RATIO=1022;SVTYPE=CNV;END=1040000;SVLEN=10000\tGT:CN\t.:1022",
                "chr14\t1200\t.\tC\t.\t.\tLowDepth\tEND=12560;REPREF=3786;RL=11360;RU=GCT;VARID=ATXN3;REPID=ATXN3\tGT:SO:REPCN:REPCI:ADSP:ADFL:ADIR:LC\t./.:./.:./.:./.:./.:./.:.:0",
                "chr14\t15000\t.\tA\t.\t.\tLowDepth\tEND=16550;REPREF=516;RL=1550;RU=GCT;VARID=ATXN2;REPID=ATXN2\tGT:SO:REPCN:REPCI:ADSP:ADFL:ADIR:LC\t./.:./.:./.:./.:./.:./.:.:0"
            ],
            "test.vcf.gz.tbi"
        ],
        "timestamp": "2023-11-28T16:32:17.674951106"
    }
}